polaris-react/src/components/Button/Button.scss

```diff
@@ -296,266 +257,238 @@
   --pc-button-color-active: var(--p-color-bg-primary-active);
   --pc-button-color-depressed: var(--p-color-bg-primary-active);
   // stylelint-enable
 
-  #{$se23} & {
-    &:not(.disabled) {
-      box-shadow: none;
+  &::before {
+    /* shine */
+    // stylelint-disable -- set primary styles
+    content: '';
+    position: absolute;
+    left: 0;
+    right: 0;
+    top: 0;
+    bottom: 0;
+    background: linear-gradient(
+      180deg,
+      rgba(255, 255, 255, 0.07) 80%,
+      rgba(255, 255, 255, 0.15) 100%
+    );
+    box-shadow: var(--p-shadow-button-primary-strong-experimental);
+    border-radius: 9px;
+    transition: opacity 75ms cubic-bezier(0.19, 0.91, 0.38, 1);
+    mix-blend-mode: luminosity;
+    // stylelint-enable
+  }
+
+  &:not(.disabled) {
+    box-shadow: none;
+  }
+
+  &:hover {
+    // stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- Button custom property
+    --pc-button-text: var(--p-color-text-inverse);
+
+    &::before {
+      box-shadow: var(--p-shadow-button-primary-strong-experimental);
+    }
+
+    svg {
+      fill: var(--p-color-text-inverse);
     }
+  }
 
+  &:active {
+    // stylelint-disable polaris/conventions/polaris/custom-property-allowed-list -- Button custom properties
+    --pc-button-text: var(--p-color-text-inverse-subdued);
+    background: var(--pc-button-color-active);
+    box-shadow: var(--p-shadow-button-inset-experimental);
+    // stylelint-enable polaris/conventions/polaris/custom-property-allowed-list
+
+    // stylelint-enable-next-line length-zero-no-unit
     &::before {
-      /* shine */
-      // stylelint-disable -- se23 temporary styles
-      content: '';
-      position: absolute;
-      left: 0;
-      right: 0;
-      top: 0;
-      bottom: 0;
+      opacity: 1;
+      // stylelint-disable-next-line -- override background
       background: linear-gradient(
         180deg,
-        rgba(255, 255, 255, 0.07) 80%,
-        rgba(255, 255, 255, 0.15) 100%
+        rgba(255, 255, 255, 0.1) 0%,
+        rgba(255, 255, 255, 0) 100%
       );
-      box-shadow: var(--p-shadow-button-primary-strong-experimental);
-      border-radius: 9px;
-      transition: opacity 75ms cubic-bezier(0.19, 0.91, 0.38, 1);
-      mix-blend-mode: luminosity;
-      // stylelint-enable
+      box-shadow: var(--p-shadow-button-primary-strong-inset-experimental);
+    }
+
+    svg {
+      fill: var(--p-color-text-inverse-subdued);
+    }
+  }
+
+  &.disabled {
+    // stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom property
+    color: var(--pc-button-text);
+    background: var(--p-color-bg-transparent-primary-disabled-experimental);
+    box-shadow: none;
+
+    &::before {
+      display: none;
+    }
+
+    svg {
+      // stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- override Spinner fill color
+      fill: var(--pc-button-text) !important;
```

Comment:
Had to include the `!important` to fix incorrect Spinner color when `loading` is true on a primary Button.